### PR TITLE
sock_merchant.py: Matching socks by color

### PR DESCRIPTION
**Pull request by SouvikGhosh05**
### Describe your change:

* [x] Add an algorithm?
* [ ] Fix a bug or typo in an existing algorithm?
* [x] Documentation change?

### Checklist:
* [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [x] This pull request is all my own work -- I have not plagiarized.
* [x] I know that pull requests will not be merged if they fail the automated tests.
* [x] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [x] All new Python files are placed inside an existing directory.
* [x] All filenames are in all lowercase characters with no spaces or dashes.
* [x] All functions and variable names follow Python naming conventions.
* [x] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [x] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [x] All new algorithms have a URL in its comments that points to Wikipedia or other similar explanation.
* [x] If this pull request resolves one or more open issues then the commit message contains `Fixes: #{$ISSUE_NO}`.

Happy to contribute in the repo.
If there's any issue arises, kindly let me know. 
Thanks!

### CODE REVIEW

**Comment by SouvikGhosh05** [comment]
@L3str4nge @cclauss kindly review it. For any issues, let me know. Thanks!

**Comment by cclauss** [comment]
Please add a URL to a description of what this algorithm is and why solving is useful.

**Comment by SouvikGhosh05** [comment]
What may be the cause of build failure?

**Comment by SouvikGhosh05** [comment]
I fixed it, probably it should now work properly. @cclauss

**Comment by SouvikGhosh05** [comment]
The error is not from my code
```
ERROR computer_vision/cnn_classification.py - tensorflow.python.framework.err...
ERROR machine_learning/lstm/lstm_prediction.py - tensorflow.python.framework....
!!!!!!!!!!!!!!!!!!! Interrupted: 2 errors during collection !!!!!!!!!!!!!!!!!!!!
======================== 1 warning, 2 errors in 21.79s =========================
Error: Process completed with exit code 2```

**Comment by cclauss** [comment]
https://github.com/TheAlgorithms/Python/pull/5762

File: digital_image_processing/edge_detection/canny.py [hunk 1]
```diff
@@ -1,6 +1,5 @@
 import cv2
 import numpy as np
-
```
Comment (cclauss) [inline digital_image_processing/edge_detection/canny.py:3]: isort will want to see this blank line between internal and external imports.

**Comment by SouvikGhosh05** [comment]
@cclauss I tried my best to fix the build. Now, you can review.

**Review by cclauss — APPROVED** [review]
LGTM

**Pull request merged**